### PR TITLE
Feature/delete households

### PR DESCRIPTION
**Pull request by sansive**
## Task
Implement delete household button. It deletes events associated with the household as well.

## Testing
Done in `EditHouseholdTest.java`

## To-Do
* Delete task list associated with the household
* Delete grocery list associated with the household
* Delete cached data associated with the household

### CODE REVIEW

File: app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java [hunk 1]
```diff
@@ -26,141 +32,135 @@ public class EditHousehold extends AppCompatActivity {
...
                     DocumentSnapshot document = task.getResult();
                     Map<String, Object> householdData = document.getData();
-                    if(householdData != null) {
+                    if (householdData != null) {
                         TextView tv = findViewById(R.id.edit_household_name);
-                        tv.setText(householdData
-                          .getOrDefault("name", "No house name")
-                          .toString()); //Ignore IDE null warning as check is done above
+                        // Ignore IDE null warning as check is done above
+                        tv.setText(householdData.getOrDefault("name", "No house name").toString());
```
Comment (Oxydeas) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java:55]: Nice catch

File: app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java [hunk 2]
```diff
@@ -26,141 +32,135 @@ public class EditHousehold extends AppCompatActivity {
...
                     }
                 });
     }
 
-    private boolean verifyEmailHasCorrectFormat(String s){
+    private boolean verifyEmailHasCorrectFormat(String s) {
```
Comment (Oxydeas) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java:60]: I think we should move this to a helper somewhere

File: app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java [hunk 3]
```diff
@@ -171,34 +171,104 @@ public void removeUser(View view) {
...
-        Intent intent = new Intent(this, HouseSelectionActivity.class);
-        startActivity(intent);
+    public void deleteCalendar(View view) {
+        firestore.collection("events")
+                .whereEqualTo("household", currentHousehold)
+                .get().addOnCompleteListener(task -> {
+            if (task.isSuccessful()) {
+                for (QueryDocumentSnapshot document : task.getResult()) {
+                    firestore.collection("events").document(document.getId()).delete()
+                            .addOnFailureListener(new OnFailureListener() {
```
Comment (Oxydeas) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java:240]: Can you replace this by a lambda ?

File: app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java [hunk 4]
```diff
@@ -171,34 +171,104 @@ public void removeUser(View view) {
...
+
+            } else {
+                Toast.makeText(getApplicationContext(), view.getContext().getString(R.string.remove_calendar_failure), Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
+
+    public void deleteHousehold(View view) {
+        firestore.collection("households").document(householdId).delete()
+                .addOnSuccessListener(new OnSuccessListener<Void>() {
```
Comment (Oxydeas) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java:256]: Can you replace this by a lambda ?

File: app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java [hunk 5]
```diff
@@ -171,34 +171,104 @@ public void removeUser(View view) {
...
+                    public void onSuccess(Void aVoid) {
+                        Toast.makeText(getApplicationContext(),
+                                view.getContext().getString(R.string.remove_household_success),
+                                Toast.LENGTH_SHORT).show();
+
+                        Intent intent = new Intent(getApplicationContext(), HouseSelectionActivity.class);
+                        startActivity(intent);
+                    }
+                })
+                .addOnFailureListener(new OnFailureListener() {
```
Comment (Oxydeas) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/house/EditHousehold.java:267]: Can you replace this by a lambda ?

**Review by Oxydeas — CHANGES_REQUESTED** [review]
Overall pretty good addition thanks! Please check the fiew comments about lambdas and get back to me if you can modify so that it looks a bit better :)

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirebaseTestsHelper.java [hunk 6]
```diff
@@ -202,6 +202,24 @@ protected static Map<String, Object> fetchHouseholdData(String houseName, Fireba
...
+        return task.getResult().getDocuments();
+    }
+
+    protected static boolean householdExists(String houseName, FirebaseFirestore db) throws ExecutionException, InterruptedException {
+        Task<DocumentSnapshot> task = db.collection("households").document(houseName).get();
+        Tasks.await(task);
+        return task.getResult().exists();
+    }
+
+    protected static boolean EventExists(String event, FirebaseFirestore db) throws ExecutionException, InterruptedException {
```
Comment (aindreias) [inline app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirebaseTestsHelper.java:217]: Is there a reason to start the method name with a capital letter here?

**Review by aindreias — APPROVED** [review]
Great code, thanks for the nice work!

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirebaseTestsHelper.java (hunk 6 above)
Comment (sansive) [reply app/src/androidTest/java/com/github/houseorganizer/houseorganizer/FirebaseTestsHelper.java:217]: No, just a typo. I will change it before merging.

**Pull request merged**